smolyay/basis.py

```diff
@@ -128,6 +137,35 @@ def __call__(self,x):
                 answer += math.comb(self._n,2*k)*((x**2 - 1)**k)*(x**(self._n-2*k))
             return answer
 
+    def derivative(self, x):
+        """Evaluate the derivative of ChebyshevFirstKind.
+
+        The first derivative of Chebyshev polynomials of first kind is
+        evaluated using the relation between Chebyshev polynomial of
+        first kind and second kind.
+
+        ..math::
+        T_n'(x) = nU_{n-1}(x)
+
+        Parameters
+        ----------
+        x: float
+            input in [-1, 1] domain.
+
+        Returns
+        -------
+        float
+            Value of the derivative of Chebyshev polynomials of first kind.
+
+        Raises
+        ------
+        ValueError
+            if input is outside the domain [-1,1].
+        """
+        if x > 1 or x < -1:
+            raise ValueError("Input is outside the domain [-1,1]")
+        return self.n*self._derivative(x)
```

Comment:
With the change above:
```suggestion
        return self.n*self._derivative_polynomial(x)
```